functionary/prompt_template/prompt_template_v2.py

```diff
@@ -0,0 +1,337 @@
+    ) -> Dict:
+        """Return delta for text_response in streaming
+
+        Args:
+            delta_text (Optional[str]): _description_
+            return_role (bool): _description_
+            finish_reason (Optional[str]): _description_
+
+        Returns:
+            Dict: _description_
+        """
+        return {
+            "delta": {
+                "content": delta_text,
+                "function_call": None,
+                "role": None if not return_role else "assistant",
+                "tool_calls": None,
+            },
+            "finish_reason": finish_reason,
+            "index": 0,
+        }
+
+    def get_recipient(self, current_text: str) -> str:
+        """Get recipient from the llm_output
+
+        Args:
+            current_text (str): _description_
+
+        Returns:
+            str: _description_
+        """
+        recipient_index = current_text.find(self.recipient_token)
+        start_index = 0
+        if recipient_index >= 0:
+            start_index = recipient_index + len(self.recipient_token)
+            
+        end_index = current_text.find(f"\n{self.content_token}")
+        return current_text[start_index:end_index].strip()
+
+    def get_chat_template_jinja(self) -> str:
+        chat_template = """{% for message in messages %}
+        {% if message['role'] == 'user' or message['role'] == 'system' %}
+            {{ '<|from|>' + message['role'] + '\n<|recipient|>all\n<|content|>' + message['content'] + '\n' }}<br>
+        {% elif message['role'] == 'tool' %}
+            {{ '<|from|>' + message['name'] + '\n<|recipient|>all\n<|content|>' + message['content'] + '\n' }}<br>
+        {% else %}
+            {% set contain_content='no'%}
+            {% if message['content'] is not none %}
+                {{ '<|from|>assistant\n<|recipient|>all\n<|content|>' + message['content'] }}<br>
+                {% set contain_content='yes'%}
+            {% endif %}
+            {% if 'tool_calls' in message and message['tool_calls'] is not none %}
+                {% for tool_call in message['tool_calls'] %}
+                    {% set prompt='<|from|>assistant\n<|recipient|>' + tool_call['function']['name'] + '\n<|content|>' + tool_call['function']['arguments'] %}
+                    {% if loop.index == 1 and contain_content == "no" %}
+                        {{ prompt }}<br>
+                    {% else %}
+                        {{ '\n' + prompt}}<br>
+                    {% endif %}
+                {% endfor %}
+            {% endif %}
+            {{ '<|stop|>\n' }}<br>
+        {% endif %}
+        {% endfor %}
+        {% if add_generation_prompt %}{{ '<|from|>assistant\n<|recipient|>' }}{% endif %}
+        """
+        chat_template = chat_template.replace("    ", "")
+        chat_template = chat_template.replace("<br>\n", "")
+        chat_template = chat_template.strip()
+        return chat_template
+
+    def update_response_state_from_delta_text(
```

Comment:
Can you still provide a docstring explaining the individual fields in `current_state` and their use?